src/pages/CommandView/form-data/get-submit-argument/get-submit-argument.ts

```diff
@@ -0,0 +1,342 @@
+    start_date: Date.parse(triggerData.interval_start_date as string),
+    end_date: Date.parse(triggerData.interval_end_date as string),
+    timezone: triggerData.interval_timezone as string,
+    jitter: triggerData.interval_jitter as number,
+    reschedule_on_finish: triggerData.interval_reschedule_on_finish as boolean,
+    seconds: 0,
+    minutes: 0,
+    hours: 0,
+    days: 0,
+    weeks: 0,
+  }
+  const intervalType = triggerData.interval as IntervalType
+
+  const trigger: IntervalTrigger = {
+    ...baseTrigger,
+    [intervalType]: triggerData.interval_num as number,
+  }
+
+  return {
+    trigger_type: type,
+    trigger: trigger,
+  }
+}
+
+const getCronTrigger = (triggerData: ObjectWithStringKeys): CombinedTrigger => {
+  const type: TriggerType = 'cron'
+  const trigger: CronTrigger = {
+    minute: triggerData.minute as string,
+    hour: triggerData.hour as string,
+    day: triggerData.day as string,
+    month: triggerData.month as string,
+    day_of_week: triggerData.day_of_week as string,
+    year: triggerData.year as string,
+    week: triggerData.week as string,
+    second: triggerData.second as string,
+    timezone: triggerData.cron_timezone as string,
+    ...('cron_jitter' in triggerData
+      ? { jitter: triggerData.cron_jitter as number }
+      : null),
+    ...('cron_start_date' in triggerData
+      ? { start_date: Date.parse(triggerData.cron_start_date as string) }
+      : null),
+    ...('cron_end_date' in triggerData
+      ? { end_data: Date.parse(triggerData.cron_end_date as string) }
+      : null),
+  }
+
+  return {
+    trigger_type: type,
+    trigger: trigger,
+  }
+}
+
+const extractTrigger = (
+  model: ObjectWithStringKeys,
+): CombinedTrigger | null => {
+  if (model && 'job' in model) {
+    const job = model.job as ObjectWithStringKeys
+
+    if (!job || !('trigger' in job)) return null
+
+    const triggerType = job.trigger as TriggerType
+
+    switch (triggerType) {
+      case 'cron':
+        return getCronTrigger(job)
+      case 'date':
+        return getDateTrigger(job)
+      case 'interval':
+        return getIntervalTrigger(job)
+      default:
+        return null
+    }
+  }
+  return null
+}
+
+const getJobPayload = (
+  model: ObjectWithStringKeys,
+  command: AugmentedCommand,
+): Job => {
+  const requestTemplate = extractRequestTemplate(model, command)
+  const optionals = extractJobOptionals(model)
+  const trigger = extractTrigger(model)
+  const name = extractName(model)
+
+  if (!name || !trigger || !requestTemplate) {
+    throw new Error('Unrecoverable error in job creation')
+  }
+
+  const job = {
+    ...name,
+    ...optionals,
+    request_template: requestTemplate,
+    ...trigger,
+    id: null,
+  }
+
+  return job
+}
+
+const getRequestPayload = (
+  model: ObjectWithStringKeys,
+  command: AugmentedCommand,
+  hasBytes = false,
+): RequestTemplate => {
+  return extractRequestTemplate(model, command, hasBytes)
+}
+
+const getSubmitArgument = (
```

Comment:
This converts the frontend model into the format the backend needs for submission.